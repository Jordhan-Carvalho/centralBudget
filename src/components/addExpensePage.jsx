import React, { Component } from "react";
import { connect } from "react-redux";
import ExpenseForm from "./expenseForm";
import { addExpense } from "../actions/expenses";

export class AddExpensePage extends Component {
  onSubmit = expense => {
    this.props.addExpense(expense);
    this.props.history.push("/");
  };

  render() {
    return (
      <React.Fragment>
        <h1>Add exprense</h1>
        <ExpenseForm onSubmit={this.onSubmit} />
      </React.Fragment>
    );
  }
}

// to simplify props.dispatch(addExpense(expense)) to props.onSubmit(expense)
const mapDispatchToProps = dispatch => {
  return {
    addExpense: expense => dispatch(addExpense(expense))
  };
};

export default connect(
  undefined,
  mapDispatchToProps
)(AddExpensePage);
