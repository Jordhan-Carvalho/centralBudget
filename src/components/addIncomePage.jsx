import React, { Component } from "react";
import { connect } from "react-redux";
import Form from "./common/form";
import { startAddIncome } from "../actions/incomes";

export class AddIncomePage extends Component {
  onSubmit = income => {
    this.props.startAddIncome(income);
    this.props.history.push("/dashboard");
  };

  render() {
    return (
      <React.Fragment>
        <h1>Add Income</h1>
        <Form onSubmit={this.onSubmit} />
      </React.Fragment>
    );
  }
}

// to simplify props.dispatch(addExpense(expense)) to props.onSubmit(expense)
const mapDispatchToProps = dispatch => {
  return {
    startAddIncome: income => dispatch(startAddIncome(income))
  };
};

export default connect(
  undefined,
  mapDispatchToProps
)(AddIncomePage);
