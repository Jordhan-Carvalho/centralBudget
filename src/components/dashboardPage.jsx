import React, { Component } from "react";
import TransChart from "./transChart";
import SavingsChart from "./savingsChart";
import ListFilters from "./common/listFilters";
import TransSumary from "./common/transSumary";

class DashboardPage extends Component {
  render() {
    return (
      <React.Fragment>
        <div className="row justify-content-center">
          <div className="col-12 boxdash">
            <TransChart height={200} />
          </div>
        </div>

        <div className="row justify-content-center">
          <div className="col-12 boxdash">
            <SavingsChart height={200} />
          </div>
        </div>

        <div className="row">
          <div className="col-12 boxdash">
            <ListFilters />
            <TransSumary type={"expense"} />
            <TransSumary type={"income"} />
            <TransSumary type={"saving"} />
          </div>
        </div>
      </React.Fragment>
    );
  }
}

export default DashboardPage;
