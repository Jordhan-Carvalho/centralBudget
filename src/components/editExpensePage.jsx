import React, { Component } from "react";
import { connect } from "react-redux";
import Form from "./common/form";
import { startEditExpense, startRemoveExpense } from "../actions/expenses";

export class EditExpensePage extends Component {
  onRemove = () => {
    this.props.startRemoveExpense(this.props.expense);
    this.props.history.push("/");
  };

  onSubmit = expense => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push("/");
  };

  render() {
    return (
      <div className="form-group">
        <h1>Edit {this.props.expense.description}</h1>
        <div className="form-group">
          <Form expense={this.props.expense} onSubmit={this.onSubmit} />
        </div>
        <div className="form-group">
          <button className="btn btn-danger" onClick={this.onRemove}>
            Remove
          </button>
        </div>
      </div>
    );
  }
}

// maps

const mapDispatchToProps = dispatch => {
  return {
    startEditExpense: (expenseID, expense) =>
      dispatch(startEditExpense(expenseID, expense)),
    startRemoveExpense: expense => dispatch(startRemoveExpense(expense))
  };
};

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(expense => {
      return expense.id === props.match.params.id;
    })
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditExpensePage);
