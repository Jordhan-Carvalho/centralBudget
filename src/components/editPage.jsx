import React, { Component } from "react";
import { connect } from "react-redux";
import ExpenseForm from "./expenseForm";
import { startEditExpense, startRemoveExpense } from "../actions/expenses";

export class EditPage extends Component {
  onRemove = () => {
    this.props.startRemoveExpense(this.props.expense);
    this.props.history.push("/");
  };

  onSubmit = expense => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push("/");
  };

  render() {
    return (
      <div>
        <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
        <button className="btn btn-primary" onClick={this.onRemove}>
          Remove
        </button>
      </div>
    );
  }
}

// maps

const mapDispatchToProps = dispatch => {
  return {
    startEditExpense: (expenseID, expense) =>
      dispatch(startEditExpense(expenseID, expense)),
    startRemoveExpense: expense => dispatch(startRemoveExpense(expense))
  };
};

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(expense => {
      return expense.id === props.match.params.id;
    })
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditPage);
