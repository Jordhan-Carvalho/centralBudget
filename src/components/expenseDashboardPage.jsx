import React, { Component } from "react";
import { Link } from "react-router-dom";
import ExpenseList from "./expenseList";
import ExpenseListFilters from "./expenseListFilters";

class ExpenseDashboardPage extends Component {
  state = {};
  render() {
    return (
      <React.Fragment>
        <h1>Hello</h1>
        <ExpenseListFilters />
        <ExpenseList />
        <Link to="/create">Add expense</Link>
        <Link to="edit">edit expense</Link>
      </React.Fragment>
    );
  }
}

export default ExpenseDashboardPage;
