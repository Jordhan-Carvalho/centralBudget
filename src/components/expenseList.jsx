import React from "react";
import { connect } from "react-redux";
import ListItem from "./common/listItem";
import selectVisible from "../selectors/visibleTrans";

export const ExpenseList = props => {
  return (
    <div>
      <h1>Expense List</h1>
      {props.expenses.length === 0 ? (
        <p>No expenses</p>
      ) : (
        <ListItem type={"expense"} expenses={props.expenses} />
      )}
    </div>
  );
};

// redux call
const mapStateToProps = state => {
  return {
    expenses: selectVisible(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
