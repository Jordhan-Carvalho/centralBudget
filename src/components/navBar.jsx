import React, { Component } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { startLogout } from "../actions/auth";

export class NavBar extends Component {
  state = {
    selected: ""
  };
  render() {
    return (
      <React.Fragment>
        <div id="cont">
          <div id="menu-fixed">
            <a href="#cont">
              <i className="material-icons back">&#xE314;</i>
            </a>
            <a href="#menu-fixed">
              <div className="logo">
                <span />
                <p>CB</p>
              </div>
              <p className="pmenu">MENU</p>
            </a>
            <hr />
            <ul className="menu">
              <Link to="/dashboard">
                <li>
                  <i className="material-icons">&#xE88A;</i>
                  <p>Home</p>
                </li>
              </Link>
              <Link to="/incomes">
                <li>
                  <i className="material-icons"> add</i>
                  <p>Incomes</p>
                </li>
              </Link>
              <Link to="/expenses">
                <li>
                  <i className="material-icons">remove</i>
                  <p>Expenses</p>
                </li>
              </Link>
              <Link to="/savings">
                <li>
                  <i className="material-icons">attach_money</i>
                  <p>Savings</p>
                </li>
              </Link>
              <Link to="/help">
                <li>
                  <i className="material-icons">&#xE8B8;</i>
                  <p>Settings</p>
                </li>
              </Link>
              <li onClick={this.props.startLogout}>
                <i className="material-icons">desktop_access_disabled</i>
                <p>Logout</p>
              </li>
            </ul>
            <i className="material-icons info">&#xE88E;</i>
          </div>
        </div>
      </React.Fragment>
    );
  }
}

// redux

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(
  undefined,
  mapDispatchToProps
)(NavBar);
