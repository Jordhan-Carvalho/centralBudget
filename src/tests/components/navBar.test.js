import React from 'react';
import { shallow } from 'enzyme';
import { NavBar } from '../../components/navBar';

test('should render NavBar correctly', () => {
    const wrapper = shallow(<NavBar startLogout={() => {}}/>);
    expect(wrapper).toMatchSnapshot();

});

test('should call startLogout on button click', () => {
    const startLogout = jest.fn();
    const wrapper = shallow(<NavBar startLogout={startLogout}/>);
    wrapper.find('li').at(5).simulate('click');
    expect(startLogout).toHaveBeenCalled();
});
