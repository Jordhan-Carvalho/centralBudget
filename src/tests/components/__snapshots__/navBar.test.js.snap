// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NavBar correctly 1`] = `
<Fragment>
  <div
    id="cont"
  >
    <div
      id="menu-fixed"
    >
      <a
        href="#cont"
      >
        <i
          className="material-icons back"
        >
          
        </i>
      </a>
      <a
        href="#menu-fixed"
      >
        <div
          className="logo"
        >
          <span />
          <p>
            CB
          </p>
        </div>
        <p
          className="pmenu"
        >
          MENU
        </p>
      </a>
      <hr />
      <ul
        className="menu"
      >
        <Link
          to="/dashboard"
        >
          <li>
            <i
              className="material-icons"
            >
              
            </i>
            <p>
              Home
            </p>
          </li>
        </Link>
        <Link
          to="/incomes"
        >
          <li>
            <i
              className="material-icons"
            >
               add
            </i>
            <p>
              Incomes
            </p>
          </li>
        </Link>
        <Link
          to="/expenses"
        >
          <li>
            <i
              className="material-icons"
            >
              remove
            </i>
            <p>
              Expenses
            </p>
          </li>
        </Link>
        <Link
          to="/savings"
        >
          <li>
            <i
              className="material-icons"
            >
              attach_money
            </i>
            <p>
              Savings
            </p>
          </li>
        </Link>
        <Link
          to="/help"
        >
          <li>
            <i
              className="material-icons"
            >
              
            </i>
            <p>
              Settings
            </p>
          </li>
        </Link>
        <li
          onClick={[Function]}
        >
          <i
            className="material-icons"
          >
            desktop_access_disabled
          </i>
          <p>
            Logout
          </p>
        </li>
      </ul>
      <i
        className="material-icons info"
      >
        
      </i>
    </div>
  </div>
</Fragment>
`;
